code/modules/cm_marines/dropship_equipment.dm

```diff
@@ -961,9 +963,17 @@
 
 	busy_winch = FALSE
 	var/fail
-	if(!linked_stretcher || linked_stretcher != old_stretcher || !is_ground_level(linked_stretcher.z))
+	if(!linked_stretcher || linked_stretcher != old_stretcher)
 		fail = TRUE
 
+	if(!is_ground_level(linked_stretcher.z)) //in case the stretcher was on a groundside dropship that flew away during our input()
+		if(istype(linked_stretcher, /obj/structure/bed/medevac_stretcher/vehicle))
+			var/obj/structure/bed/medevac_stretcher/vehicle/v_stretcher = linked_stretcher
+			if(!v_stretcher.vehicle || v_stretcher.vehicle.health < (initial(v_stretcher.vehicle.health) * 0.5) || !is_ground_level(v_stretcher.vehicle.z) || v_stretcher.vehicle.next_move > world.time)
+				fail = TRUE
+		else
+			fail = TRUE
```

Comment:
same here (also all 3 of these use the same lines, all the more reason to make a proc!)

Reply:
Made a separate proc in stretcher.